R/touchstone_countries.R

```diff
@@ -0,0 +1,99 @@
+###############################################################################
+
+extract_touchstone_country <- function(e, path, con) {
+  csv <- e$touchstone_countries_csv
+  if (is.null(csv)) {
+    return(e)
+  }
+
+  if (!setequal(names(csv), c("touchstone", "diseases", "countries"))) {
+    stop("Invalid columns in touchstone_country.csv")
+  }
+
+  diseases <- unlist(lapply(csv$diseases, split_semi))
+  countries <- unlist(lapply(csv$countries, split_semi))
+
+  c(e, list(
+    tc_disease = db_get(con, "disease", "id", unique(diseases)),
+    tc_country = db_get(con, "country", "id", unique(countries)),
+    db_touchstone_country = db_get(con, "touchstone_country", "touchstone",
+                                   unique(csv$touchstone))
+  ))
+}
+
+test_extract_touchstone_country <- function(e) {
+  csv <- e$touchstone_countries_csv
+  if (is.null(csv)) {
+    return()
+  }
+
+  diseases <- lapply(csv$diseases, split_semi)
+  countries <- lapply(csv$countries, split_semi)
+
+  if (any(lengths(diseases) < 1)) {
+    stop("Empty disease entry in touchstone_country")
+  }
+
+  if (any(lengths(countries) < 1)) {
+    stop("Empty country entry in touchstone_country")
+  }
+
+  testthat::expect_true(all(unique(unlist(countries)) %in% e$tc_country$id),
+    label = "All countries in touchstone_country are recognised")
+
+  testthat::expect_true(all(unique(unlist(diseases)) %in% e$tc_disease$id),
+    label = "All diseases in touchstone_country are recognised")
+
+}
+
+###############################################################################
+
+transform_touchstone_country <- function(e) {
+
+  # CSV Format: touchstone,disease1;disease2,country1;country2;country3
+
+  csv <- e$touchstone_countries_csv
+  if (is.null(csv)) {
+    return(list())
+  }
+
+  diseases <- lapply(csv$diseases, split_semi)
+  countries <- lapply(csv$countries, split_semi)
+
+  # Build a table that multiplies out countries and diseases...
+  # for each touchstone.
+
+  expand_countries <- unlist(lapply(1:nrow(csv), function(x) {
```

Comment:
small thing, but can we avoid `1:nrow(csv)` here please?

Reply:
Done - good spot. (Would have failed on country CSV with a header but no rows)